src/logger/src/metrics.rs

```diff
@@ -489,6 +498,33 @@ pub struct SerialDeviceMetrics {
     pub write_count: SharedMetric,
 }
 
+/// Metrics related to signals.
+#[derive(Default, Serialize)]
+pub struct SignalMetrics {
+    /// Number of times that SIGBUS was handled.
+    pub sigbus: SharedMetric,
+    /// Number of times that SIGSEGV was handled.
+    pub sigsegv: SharedMetric,
+}
+
+#[cfg(target_arch = "x86_64")]
+/// Metrics related to snapshots.
+// These store the duration of creating/loading a snapshot.
+// If there are more than one `/snapshot/create` request in a minute
+// (until the metrics are flushed), only the duration of the last
+// snapshot creation is stored in the metric. If the user is interested
+// in all the durations, a `FlushMetrics` request should be sent after
+// each `create` request.
+#[derive(Default, Serialize)]
+pub struct SnapshotMetrics {
+    /// Measures the snapshot full create time in microseconds.
+    pub full_create_us: SharedMetric,
+    /// Measures the snapshot diff create time in microseconds.
+    pub diff_create_us: SharedMetric,
+    /// Measures the snapshot load time in microseconds.
+    pub load_us: SharedMetric,
```

Comment:
The way I understand this metric is that it measures how much it takes for us from receiving the API call to when the VMM is built.

We also need a metric which measures the resume time. We could then do load_us + resume_use = restore_time in the dashboard.

The problem with this is that it does not show how much time is lost between load and resume. Another option would be to compute the resume_us metric to span from load api call to resume completion.

Reply:
With the current implementation, resuming the microVM from snapshot is completely independent of the load operation. If we measure them separately and then do `load_us + resume_us`, the result won't tell us what we want to know, i.e. how much does it take to resume from snapshot. If we want to define the `resume_us` metric as spanning from load api call to resume completion, we kind of assume that the user will resume the vm immediately after loading and, even if it's not the most expected scenario, resuming can happen also after a few minutes (and basically anytime) and the metric won't be that useful.

Anyway, this metric (`resume_us`) would make sense only if we include indeed the API flow too(as you've suggested in another comment here), but since these operations are separate (and moreover, one of them is pre boot and the after post boot), I'm not a big fan of combining them in a metric :(. I also expect the code to get pretty ugly and counter-intuitive (among others, we will have to check in the resume function if the microVM was started from a snapshot, if we don't do that , `resume_us` will be modified also when resuming after a `create` and this way the metric will have a different meaning). I expect this metric would make more sense when resuming from a snapshot in one shot and not with 2 separate APIs.

Reply:
Resuming in one shot is possible if the REST client reuses the connection and does not create one for each request: Connection: keep-alive header.